src/index.ts

```diff
@@ -1,8 +1,8 @@
 "use strict";
 
-const fs = require('fs-extra');
-const path = require('path');
-const nodefs = require('fs');
+import fs = require('fs-extra');
+import path = require('path');
+import nodefs = require('fs');
```

Comment:
Converting these require statements to imports allows typescript to not assume their type to be `any`. This caused a number of compilation errors which are also cleaned up in this PR.